index.js

```diff
@@ -1102,7 +1070,7 @@ class Hypercore extends EventEmitter {
 
   _getEncryptionProvider (encryptionKey, block) {
     if (!encryptionKey) return null
-    return new DefaultEncryption(encryptionKey, block)
+    return new DefaultEncryption(encryptionKey, this.key, { block, compat: this.core.compat })
```

Comment:
is this called just in time? ie just before decrypt? cause it only knows the compat flag post replication